### ISSUE TITLE
hss improperly depends on mpfs-amp-examples

### ISSUE THREAD

**Issue by codusnocturnus** [polarfire-soc/meta-polarfire-soc-yocto-bsp#30]
hss.bb contains the following line
```
# Strict dependency
do_configure[depends] += "u-boot-mpfs:do_deploy mpfs-amp-examples:do_deploy"
```

It seems like mpfs-amp-examples should be tied specifically to the icicle-kit-es-amp MACHINE.
```
# Strict dependency
do_configure[depends] += "u-boot-mpfs:do_deploy"
do_configure_icicle-kit-es-amp[depends] += "mpfs-amp-examples:do_deploy"
```

**Comment by vfalanis** [comment]
Hi @codusnocturnus,

A new way to solve strict dependencies in the hss recipe based on the machine type is scheduled for our next release.

**Comment by lhanlyu** [comment]
Fixed in the release 2020.11
